frontend/src/components/NavBar.js

```diff
@@ -185,13 +290,35 @@ const NavBar = () => {
                 <FaCaretDown />
               </DropdownButton>
               <DropdownContent>
-                <DropdownItem to={`/profile/${username}`}>My Profile</DropdownItem>
-                <DropdownItem to={`/edit-profile/${username}`}>Edit Profile</DropdownItem>
-                <LogoutItem onClick={handleLogout}>Logout</LogoutItem>
+                <DropdownItem
+                  onClick={handleHideMenuList}
+                  to={`/profile/${username}`}
+                >
+                  My Profile
+                </DropdownItem>
+                <DropdownItem
+                  onClick={handleHideMenuList}
+                  to={`/edit-profile/${username}`}
+                >
+                  Edit Profile
+                </DropdownItem>
+                <LogoutItem
+                  onClick={() => {
+                    handleHideMenuList();
+                    handleLogout();
+                  }}
+                >
+                  Logout
+                </LogoutItem>
```

Comment:
In `LogoutItem`, consider creating a function that combines these actions instead of adding `handleHideMenuList `and `handleLogout` in an inline `onClick` event. This would be more efficient and consistent.